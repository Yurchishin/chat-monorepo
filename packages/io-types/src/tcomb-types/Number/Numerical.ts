import t from 'tcomb-validation'
import R from '@chat/ramda'

export const Numerical = (value: number) => t.refinement(
    t.Number,
    R.equals(value),
    'Numerical',
)
