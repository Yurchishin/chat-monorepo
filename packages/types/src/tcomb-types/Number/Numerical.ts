import t, { Irreducible } from 'tcomb-validation'
import R from '@chat/ramda'

export const Numerical = <T>(value: T, name?: string): Irreducible<T> => t.irreducible(
    name || 'Numerical',
    R.equals(value),
)
