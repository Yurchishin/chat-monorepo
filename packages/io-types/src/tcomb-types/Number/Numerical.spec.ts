import {Numerical} from './Numerical'
import {expectTypeSuccess, expectTypeFailure} from "../../tests"

describe('Numerical(5)', () => {
	const Number5 = Numerical(5)

	expectTypeSuccess(Number5, 5)
	expectTypeFailure(Number5, 4)
	expectTypeFailure(Number5, 6)
	expectTypeFailure(Number5, '1')
})

