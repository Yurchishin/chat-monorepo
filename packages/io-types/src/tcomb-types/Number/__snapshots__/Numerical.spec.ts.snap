// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Numerical(5) expectTypeFailure(1): TYPE_FAILURE::Numerical(1) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "1",
      "expected": [Function],
      "message": "Invalid value \\"1\\" supplied to Number",
      "path": Array [],
    },
  ],
  "value": "1",
}
`;

exports[`Numerical(5) expectTypeFailure(4): TYPE_FAILURE::Numerical(4) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": 4,
      "expected": [Function],
      "message": "Invalid value 4 supplied to Numerical",
      "path": Array [],
    },
  ],
  "value": 4,
}
`;

exports[`Numerical(5) expectTypeFailure(6): TYPE_FAILURE::Numerical(6) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": 6,
      "expected": [Function],
      "message": "Invalid value 6 supplied to Numerical",
      "path": Array [],
    },
  ],
  "value": 6,
}
`;
